File: backend/backend/src/main/java/com/moovie/backend/data/repository/ProfileRepositoryImpl.java
package com.moovie.backend.data.repository;

import com.moovie.backend.domain.model.*;
import com.moovie.backend.domain.repository.ProfileRepository;
import org.springframework.stereotype.Repository;

import java.util.Collections;
import java.util.List;
import java.util.Map;

@Repository
public class ProfileRepositoryImpl implements ProfileRepository {

    private static final Map<String, PublicProfile> MOCK_PROFILES = Map.of(
            "user-001", new PublicProfile(
                    "user-001", "Alice", "AL", "Film enthusiast and aspiring critic.",
                    List.of(
                            new ProfileWatchedMovie(550, "Fight Club", "/pB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg", 4.5),
                            new ProfileWatchedMovie(680, "Pulp Fiction", "/d5iIlFn5s0ImszYzBPb8JPIfbXD.jpg", 5.0)
                    ),
                    List.of(new ProfileUser("user-002", "Bob", "BO")),
                    List.of(new ProfileUser("user-003", "Carol", "CA")),
                    List.of(new ProfileFavoriteMovie(550, "Fight Club", "/pB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg")),
                    List.of(new ProfileRecentActivity("reviewed", "Fight Club", "2024-01-15")),
                    List.of(new ProfileWatchlistItem(238, "The Godfather", "/3bhkrj58Vtu7enYsRolD1fZdja1.jpg"))
            ),
            "user-002", new PublicProfile(
                    "user-002", "Bob", "BO", "Casual movie watcher.",
                    List.of(new ProfileWatchedMovie(238, "The Godfather", "/3bhkrj58Vtu7enYsRolD1fZdja1.jpg", 4.0)),
                    Collections.emptyList(),
                    List.of(new ProfileUser("user-001", "Alice", "AL")),
                    Collections.emptyList(),
                    Collections.emptyList(),
                    Collections.emptyList()
            )
    );

    @Override
    public UserProfile getUserProfile() {
        return new UserProfile(
                null,
                "currentUser",
                "Movie lover",
                Collections.emptyList(),
                Collections.emptyList(),
                Collections.emptyList()
        );
    }

    @Override
    public void updateUserProfile(UserProfile profile) {
        // TODO: persist to database
    }

    @Override
    public PublicProfile getPublicProfile(String userId) {
        return MOCK_PROFILES.getOrDefault(userId, new PublicProfile(
                userId, "Unknown", "??", null,
                Collections.emptyList(), Collections.emptyList(), Collections.emptyList(),
                Collections.emptyList(), Collections.emptyList(), Collections.emptyList()
        ));
    }
}

File: src/main/kotlin/org/mobyle/data/repository/ProfileRepositoryImpl.kt
package org.mobyle.data.repository

import org.mobyle.domain.model.*
import org.mobyle.domain.repository.ProfileRepository

class ProfileRepositoryImpl : ProfileRepository {

    companion object {
        private val mockProfiles = mapOf(
            "user-001" to PublicProfile(
                id = "user-001",
                displayName = "Alice Martins",
                initials = "AM",
                bio = "Film enthusiast and critic",
                moviesWatched = listOf(
                    ProfileWatchedMovie(550, "Fight Club", "/pB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg"),
                    ProfileWatchedMovie(278, "The Shawshank Redemption", "/q6725aR8Zs4IwsER1POIPZeroA8.jpg"),
                    ProfileWatchedMovie(238, "The Godfather", "/3bhkrj58Vtu7enYsRolD1fZQeQA.jpg")
                ),
                following = listOf(
                    ProfileUser("user-002", "Bruno Carvalho", "BC"),
                    ProfileUser("user-003", "Camila Torres", "CT")
                ),
                followers = listOf(
                    ProfileUser("user-004", "Diego Ferreira", "DF"),
                    ProfileUser("user-005", "Elena Souza", "ES")
                ),
                favoriteMovies = listOf(
                    ProfileFavoriteMovie(550, "Fight Club"),
                    ProfileFavoriteMovie(278, "The Shawshank Redemption"),
                    ProfileFavoriteMovie(238, "The Godfather")
                ),
                recentActivities = listOf(
                    ProfileRecentActivity("watched", "Inception", "2 days ago"),
                    ProfileRecentActivity("reviewed", "The Dark Knight", "5 days ago"),
                    ProfileRecentActivity("added_to_watchlist", "Parasite", "1 week ago")
                ),
                watchlist = listOf(
                    ProfileWatchlistItem(550, "Interstellar"),
                    ProfileWatchlistItem(278, "The Matrix"),
                    ProfileWatchlistItem(238, "Blade Runner 2049")
                )
            ),
            "user-002" to PublicProfile(
                id = "user-002",
                displayName = "Bruno Carvalho",
                initials = "BC",
                bio = "Movie geek, always discovering new films",
                moviesWatched = listOf(
                    ProfileWatchedMovie(278, "The Shawshank Redemption", "/q6725aR8Zs4IwsER1POIPZeroA8.jpg"),
                    ProfileWatchedMovie(238, "The Godfather", "/3bhkrj58Vtu7enYsRolD1fZQeQA.jpg")
                ),
                following = listOf(ProfileUser("user-001", "Alice Martins", "AM")),
                followers = listOf(
                    ProfileUser("user-001", "Alice Martins", "AM"),
                    ProfileUser("user-003", "Camila Torres", "CT")
                ),
                favoriteMovies = listOf(
                    ProfileFavoriteMovie(278, "The Shawshank Redemption")
                ),
                recentActivities = listOf(
                    ProfileRecentActivity("reviewed", "The Godfather Part II", "3 days ago")
                ),
                watchlist = listOf(
                    ProfileWatchlistItem(12, "Finding Nemo")
                )
            )
        )
    }

    override suspend fun getUserProfile(): UserProfile {
        return UserProfile(
            username = "@filmfan42",
            bio = "Film enthusiast and movie critic",
            photoUrl = "https://api.example.com/avatars/user-001.jpg"
        )
    }

    override suspend fun updateUserProfile(profile: UserProfile) {
        // TODO: Persist to database
    }

    override suspend fun getPublicProfile(userId: String): PublicProfile {
        return mockProfiles[userId] ?: PublicProfile(
            id = userId,
            displayName = "Unknown User",
            initials = "??"
        )
    }
}
